opal/mca/common/ucx/common_ucx.c

```diff
@@ -52,6 +83,7 @@ OPAL_DECLSPEC void opal_common_ucx_mca_deregister(void)
     if (opal_common_ucx.registered) {
         return;
     }
+    opal_mem_hooks_unregister_release(opal_common_ucx_mem_release_cb);
```

Comment:
is it going to remove memory hooks for the first component that is used? what if there are several of them?

Reply:
here is reference counter at line 83:
```
opal_common_ucx.registered--;
if (opal_common_ucx.registered) {
    return;
}
```
uninitialization will be executed on last deregister only (assuming that numbers of register and deregister are same)

Reply:
BTW is it possible we can have race condition on the counter? in case MPI_Win_create is called from multiple threads for example?

Reply:
should not be issue because pml will be initialized first and all next initializations will not take effect (and for OSC initialization it will be executed in common OSC initialization stack)